### PR TITLE
Switched to pflag for argument handling

### PR DESCRIPTION
**Pull request by zom-ponks**
Switched to pflag (https://github.com/spf13/pflag), and did a code cleanup + definitions update

### CODE REVIEW

File: args.go [hunk 1]
```diff
@@ -22,39 +23,26 @@ func parseArguments() {
...
+	pflag.StringVarP(&parsedConf.ConfigYMLPath, "configyml", "yml", parsedConf.ConfigYMLPath, "Set config.yml path")
+	pflag.StringVarP(&parsedConf.XMLCachePath, "xmlcache", "xml", parsedConf.XMLCachePath, "XML cache location")
 	// TODO: short ones for these as well?
-	flag.IntVar(&parsedConf.DLTimeout, "timeout", parsedConf.DLTimeout, "Download timeout (in seconds)")
-	flag.IntVar(&parsedConf.DLRetries, "retry", parsedConf.DLRetries, "Number of download retries")
-
-	flag.BoolVar(&parsedConf.color, "nocolor", true, "Disable output color")
+	pflag.IntVar(&parsedConf.DLTimeout, "timeout", parsedConf.DLTimeout, "Download timeout (in seconds)")
+	pflag.IntVar(&parsedConf.DLRetries, "retry", parsedConf.DLRetries, "Number of download retries")
+	pflag.BoolVar(&parsedConf.color, "nocolor", true, "Disable output color")
```
Comment (John-Gee) [inline args.go:43]: It's a little late here so apologies if I'm being stupid, but it seems weird that nocolor defaulting to true is saved in a variable named color. Is it ever reversed somewhere? Or is nocolor=true not actually disabling color?

File: args.go [hunk 2]
```diff
@@ -22,39 +23,26 @@ func parseArguments() {
...
-
-	flag.StringVar(&parsedConf.XMLCachePath, "xmlcache", parsedConf.XMLCachePath, "XML cache location")
-	flag.StringVar(&parsedConf.XMLCachePath, "xml", parsedConf.XMLCachePath, "XML cache location")
+	// this needs to be implemented either with the pflag.Value interface, or just straight string parsing
+	//pflag.StringVar(&parsedConf.verbosity, "verbosity", "Output verbosity, accepted values: error, warning, info, debug")
 
+	pflag.StringVarP(&parsedConf.Rpcs3Path, "rpcs3path", "rpcs", parsedConf.Rpcs3Path, "Set RPCS3 path")
+	pflag.StringVarP(&parsedConf.PkgDLPath, "dlpath", "dl", parsedConf.PkgDLPath, "Set download path")
+	pflag.StringVarP(&parsedConf.ConfigYMLPath, "configyml", "yml", parsedConf.ConfigYMLPath, "Set config.yml path")
+	pflag.StringVarP(&parsedConf.XMLCachePath, "xmlcache", "xml", parsedConf.XMLCachePath, "XML cache location")
```
Comment (John-Gee) [inline args.go:39]: Are the 2nd strings supposed to be the short ones? If so I am not convinced of the utility. "rpcs" is not really short, maybe it's more the long that is too long?

I don't know, maybe I'm too used to the short ones being only one char.

File: args.go [hunk 3]
```diff
@@ -22,39 +23,26 @@ func parseArguments() {
...
-	flag.StringVar(&parsedConf.PkgDLPath, "dlpath", parsedConf.PkgDLPath, "Set download path")
-	flag.StringVar(&parsedConf.PkgDLPath, "dl", parsedConf.PkgDLPath, "Set download path")
-
-	flag.StringVar(&parsedConf.ConfigYMLPath, "configyml", parsedConf.ConfigYMLPath, "Set config.yml path")
-	flag.StringVar(&parsedConf.ConfigYMLPath, "yml", parsedConf.ConfigYMLPath, "Set config.yml path")
-
-	flag.StringVar(&parsedConf.XMLCachePath, "xmlcache", parsedConf.XMLCachePath, "XML cache location")
-	flag.StringVar(&parsedConf.XMLCachePath, "xml", parsedConf.XMLCachePath, "XML cache location")
+	// this needs to be implemented either with the pflag.Value interface, or just straight string parsing
+	//pflag.StringVar(&parsedConf.verbosity, "verbosity", "Output verbosity, accepted values: error, warning, info, debug")
```
Comment (John-Gee) [inline args.go:34]: IntVar could do too, I think that's what is commonly used, although less clear.

**Review by John-Gee — CHANGES_REQUESTED** [review]

**Comment by zom-ponks** [comment]
This needs reworking, I'll do that.

**Pull request closed without merging**